ocaml/libs/http-lib/http_svr.ml

```diff
@@ -367,7 +367,18 @@ let request_of_bio_exn ~proxy_seen ~read_timeout ~total_timeout ~max_length bio
              match Astring.String.fields ~empty:false header with
              | [meth; uri; version] ->
                  (* Request-Line   = Method SP Request-URI SP HTTP-Version CRLF *)
-                 let uri, query = Http.parse_uri uri in
+                 let uri_t = Uri.of_string uri in
+                 let uri = Uri.path uri_t in
```

Comment:
Is this `uri` the same as the one on L368? If so, why converting it back?

Reply:
I would trust `Uri` to create standard-conforming URLs. One caveat though and why this needs testing:

```
utop # let t = Uri.of_string "https://foo.bar" in Uri.path t;;
- : string = ""
```

This might be surprising and you could have expected "/".

Reply:
> Is this `uri` the same as the one on L368? If so, why converting it back?

L368's `uri` is the full path+query.
Here it's split into only the path, L381 further gets only the query part.

Couldn't come up with a more precise name so I just shadowed the other (as it was done before)